### DISCUSSION TITLE
for API methods which return templates.TemplateResponse , want to know how to write testcases for them

### DISCUSSION THREAD

**Discussion by praveenRI007** [fastapi/fastapi#10395] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get("/", response_class=HTMLResponse)
async def read_all_by_user(request: Request, db: Session = Depends(get_db)):

    user = await get_current_user(request)
    if user is None:
        return RedirectResponse(url="/auth", status_code=status.HTTP_302_FOUND)

    todos = db.query(models.Todos).filter(models.Todos.owner_id == user.get("id")).all()

    return templates.TemplateResponse("home.html", {"request": request, "todos": todos, "user": user})
```

### Description

need to write test case method for methods which return template response

### Operating System

Windows

### FastAPI Version

0.95.0

### Pydantic Version

1.10.5

### Python Version

3.9

**Comment by jgould22** [comment, marked as answer]
There is a tiny bit about testing here in the starlette docs [here](https://www.starlette.io/templates/)

Tests using the test client for endpoints that return a template response will add the response.context and response.template to the response in addition to the rest of the response.

A template response is going to return to you rendered html so you can inspect it and make sure it conforms to whatever your expectations are.
